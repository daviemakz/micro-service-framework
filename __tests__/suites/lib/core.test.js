'use strict';

// Import system components
import Core, { uniqueArray, getRandomElements } from './../../../dist/lib/core';

// Test suite
describe('dist/lib/db', () => {
  describe('uniqueArray()', () => {
    test('to match snapshot', () => {
      expect(uniqueArray([1, 1, 1, 2, 3, 4, 5])).toMatchSnapshot();
    });
  });
  describe('getRandomElements()', () => {
    test('to match snapshot', () => {
      expect(
        Array.isArray(getRandomElements([1, 1, 1, 2, 3, 4, 5], 2))
      ).toMatchSnapshot();
    });
  });
  describe('EXPORTS: Core', () => {
    test('to match snapshot', () => {
      expect(Core).toMatchSnapshot();
    });
  });
});
