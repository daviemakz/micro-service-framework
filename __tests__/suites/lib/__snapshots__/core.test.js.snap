// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dist/lib/db EXPORTS: Core to match snapshot 1`] = `
Object {
  "changeInstances": [Function],
  "end": [Function],
  "getRandomElements": [Function],
  "storage": [Function],
  "uniqueArray": [Function],
}
`;

exports[`dist/lib/db getRandomElements() to match snapshot 1`] = `true`;

exports[`dist/lib/db uniqueArray() to match snapshot 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;
